### PR TITLE
firmware/attestation: better logs on failure

### PR DESCRIPTION
**Pull request by benma**

### CODE REVIEW

File: api/firmware/attestation.go [hunk 1]
```diff
@@ -78,23 +79,33 @@ func (device *Device) performAttestation() (bool, error) {
...
-	challengeSignature = response[:64]
+	bootloaderHash, rsp = rsp[:32], rsp[32:]
+	devicePubkeyBytes, rsp = rsp[:64], rsp[64:]
+	certificate, rsp = rsp[:64], rsp[64:]
+	rootPubkeyIdentifier, rsp = rsp[:32], rsp[32:]
+	challengeSignature = rsp[:64]
 
 	rootPubkeyInfo, ok := attestationPubkeys[hex.EncodeToString(rootPubkeyIdentifier)]
 	if !ok {
+		device.log.Error(fmt.Sprintf(
```
Comment (x1ddos) [inline api/firmware/attestation.go:104]: That's a funny log.Error method. Consider switching to stdlib logger in the future.

File: api/firmware/attestation.go [hunk 2]
```diff
@@ -78,23 +79,33 @@ func (device *Device) performAttestation() (bool, error) {
 	if err != nil {
 		return false, err
```
Comment (x1ddos) [inline api/firmware/attestation.go:80]: Don't you want to log the error here as well?

File: api/firmware/attestation.go [hunk 3]
```diff
@@ -78,23 +79,33 @@ func (device *Device) performAttestation() (bool, error) {
...
 	rootPubkeyInfo, ok := attestationPubkeys[hex.EncodeToString(rootPubkeyIdentifier)]
 	if !ok {
+		device.log.Error(fmt.Sprintf(
+			"could not find root pubkey. challenge=%x, response=%x, identifier=%x",
+			challenge,
+			response,
+			rootPubkeyIdentifier), nil)
 		return false, nil
 	}
 	if rootPubkeyInfo.acceptedBootloaderHashHex != "" {
```
Comment (x1ddos) [inline api/firmware/attestation.go:111]: And here, log too?

**Review by x1ddos — APPROVED** [review]
LGTM but I also see a couple `panic(errp.WithStack(err))`. I think you'll want to replace those with log because a panic may not necessarily end up in the log file if the whole app crashes.

File: api/firmware/attestation.go (hunk 2 above)
Comment (benma) [reply api/firmware/attestation.go:80]: Here the error is returned and logged in the callsite.

Comment (benma) [reply api/firmware/attestation.go:81]: added log anyway

File: api/firmware/attestation.go (hunk 3 above)
Comment (benma) [reply api/firmware/attestation.go:112]: added

**Pull request merged**